### PR TITLE
[mle] add missing defs when LONG_ROUTES is enabled

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Review by abtink — APPROVED** [review]
👍

File: src/core/thread/mle_tlvs.hpp [hunk 1]
```diff
@@ -426,6 +426,16 @@ OT_TOOL_PACKED_BEGIN
 class RouteTlv : public Tlv, public TlvInfo<Tlv::kRoute>
 {
 public:
+    enum
+    {
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+        kLinkAcceptMaxRouters       = 3,
```
Comment (simonlingoogle) [inline src/core/thread/mle_tlvs.hpp:432]: nit: Now these values are defined twice. Move them out of `class RouteTlv`?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Pull request merged**